File: 2num.py
import sys
from collections import defaultdict
imap = {}
jmap = {}
I = defaultdict(lambda:0)
J = 0
vocab = open("vocab", 'w')
for file in sys.argv[1:]:
  out = open(file+"n", 'w')
  for line in open(file):
    item,feat,val = line.split()
    if (file,item) not in imap:
      I[file] += 1
      imap[file,item] = I[file]
    if feat not in jmap:
      J += 1
      jmap[feat] = J
      print>>vocab, feat
    print>>out, imap[file,item], jmap[feat], val
  out.close()

File: 2num.pl
open(VOCAB, ">vocab");
foreach $file (@ARGV) {
  open(IN, "<$file");
  open(OUT, ">${file}n");
  while (<IN>) {
    ($item,$feat,$val) = split(" ");
    if (!$imap{"$file $item"}) {
      $imap{"$file $item"} = ++$I{$file};
    }
    if (!$jmap{$feat}) {
      $jmap{$feat} = ++$J;
      print VOCAB $feat . "\n";
    }
    print OUT $imap{"$file $item"} . " $jmap{$feat} $val\n";
  }
  close(OUT);
}
